activerecord/lib/active_record/table_metadata.rb

```diff
@@ -12,13 +12,13 @@ def initialize(klass, arel_table, association = nil)
     def resolve_column_aliases(hash)
       # This method is a hot spot, so for now, use Hash[] to dup the hash.
       #   https://bugs.ruby-lang.org/issues/7166
-      hash = Hash[hash]
-      hash.keys.grep(Symbol) do |key|
-        if klass.attribute_alias? key
-          hash[klass.attribute_alias(key)] = hash.delete key
+      new_hash = Hash[hash]
```

Comment:
I benchmark this locally and got slower results. Probably due to the extra assignment to the hash.

```
Calculating -------------------------------------
              Hash[]   125.856k i/100ms
                 dup    78.227k i/100ms
                each    92.295k i/100ms
-------------------------------------------------
              Hash[]      3.053M (± 9.6%) i/s -     15.103M
                 dup      1.291M (± 4.5%) i/s -      6.493M
                each      1.732M (±15.1%) i/s -      8.399M

Comparison:
              Hash[]:  3052908.4 i/s
                each:  1731798.9 i/s - 1.76x slower
                 dup:  1290846.2 i/s - 2.37x slower
``` 

Test failures from Active Job